src/session/session_api.c

```diff
@@ -850,6 +851,78 @@ err:	API_END_RET(session, ret);
 }
 
 /*
+ * __session_transaction_sync --
+ *	WT_SESSION->transaction_sync method.
+ */
+static int
+__session_transaction_sync(WT_SESSION *wt_session, const char *config)
+{
+	WT_CONFIG_ITEM cval;
+	WT_CONNECTION_IMPL *conn;
+	WT_DECL_RET;
+	WT_LOG *log;
+	WT_SESSION_IMPL *session;
+	struct timespec end, now;
+	uint64_t wait_secs;
+
+	session = (WT_SESSION_IMPL *)wt_session;
+	conn = S2C(session);
+	/*
+	 * If logging is not enabled there is nothing to do.
+	 */
+	if (!FLD_ISSET(conn->log_flags, WT_CONN_LOG_ENABLED))
+		return (0);
```

Comment:
All other subsystem specific methods that may be called and the subsystem is not enabled return zero.  This usage is consistent.  (Two examples being `WT_SESSION::log_printf` and `WT_CONNECTION::async_flush`.)